### DISCUSSION TITLE
valid_increment for GenEnsProd

### DISCUSSION THREAD

**Discussion by BinyuWang-NOAA** [dtcenter/METplus#2414] [Configuration] [answered]
Hi,
I have a question for defining the value of valid_increment for GenEnsProd,
I try to verify HREF 24h precipitation versus CCPA, I first combined 6H CCPA to 24H and combined 1H HREF to 24H (using PcpCombine for both).  There is no problem with the first step. Then I tried to get ensemble products by applying  GenEnsprod and EnsembleStat using the combined *nc files as input. Here is the question, what value should I set for valid_increment?  For test, I tried to use 1H, 6H, 1D, it all works and the output are the same. But if I just remove it, the run will fail. So is "valid_increment" used at all? For me, I think "valid_increment" should be set to 1d b/c the inputs are 24h, is that correct?

Thank you.

**Comment by CPKalb** [comment]
Hi:

For your second step, you would want a valid_increment of 1D.  Are you running this second step in the same configuration file as the first step?  If so, there may be other considerations for setting this.  

Valid_increment is used and necessary for the run.  I'm guessing when you used 1H or 6H, there were errors in your run due to timing issues but that it still processed the data available, which is why the output is the same in all these cases.  However, I would need to see a log file to verify that this is correct.

Christina

**Reply by BinyuWang-NOAA** [reply]
HI Christina,

Thank you for the reply. I attached the log file for "GenEnsProd" using 1H, 3H and 1D, let me know if further information is needed.

**Reply by BinyuWang-NOAA** [reply]
[metplus.log.20231108233107_6H.txt](https://github.com/dtcenter/METplus/files/13315290/metplus.log.20231108233107_6H.txt)
[Uploading metplus.log.20231110034853_1d.txt…]()

**Comment by BinyuWang-NOAA** [comment]
[metplus.log.20231108233107_6H.txt](https://github.com/dtcenter/METplus/files/13315277/metplus.log.20231108233107_6H.txt)
[metplus.log.20231110034853_1d.txt](https://github.com/dtcenter/METplus/files/13315278/metplus.log.20231110034853_1d.txt)
[metplus.log.20231109025745_1H.txt](https://github.com/dtcenter/METplus/files/13315279/metplus.log.20231109025745_1H.txt)

**Comment by BinyuWang-NOAA** [comment]
[PcpCombine_fcstHREF_APCP24h.conf.txt](https://github.com/dtcenter/METplus/files/13338848/PcpCombine_fcstHREF_APCP24h.conf.txt)
[GenEnsProd_fcstHREF_obsCCPA24h.conf.txt](https://github.com/dtcenter/METplus/files/13338849/GenEnsProd_fcstHREF_obsCCPA24h.conf.txt)
The files above are the config file I used for Pcp_combine and GenEnsProd. Thank you for help.

**Comment by georgemccabe** [comment, marked as answer]
Hi @BinyuWang-NOAA, thank you for sending the configuration files. For both the PCPCombine and GenEnsProd configurations, you are using the same value for begin and end time (INIT_BEG == INIT_END for PCPCombine, VALID_BEG == VALID_END for GenEnsProd). In this case, the increment value is not required, because only a single run time (init or valid) is bring processed.

```
VALID_BEG = {ENV[vday]}12
VALID_END = {ENV[vday]}12
```

In the GenEnsProd configuration, you are referencing an environment variable called `valid_increment` that you have defined in your environemnt/script to set the value of `VALID_INCREMENT` in the METplus configuration.

`VALID_INCREMENT = {ENV[valid_increment]}`

Removing the above line from your METplus configuration file should result in a successful run that doesn't require you to set a value for the `valid_increment` environment variable.

If the `VALID_INCREMENT` variable is unset, it uses a default value that will allow the run to succeed. However, if you do not set the `valid_increment` environment variable, your METplus run will fail because it expects that the variable is set in your environment. If you set that environment variable to an empty string, METplus will also fail because it tries to interpret the empty string as a time interval. I could enhance the code for future versions of METplus to properly handle an empty string value for the increment so it doesn't crash.

**Reply by BinyuWang-NOAA** [reply]
Thank you, George.
